Sources/MuxUploadSDK/InternalUtilities/ChunkedFile.swift

```diff
@@ -39,11 +39,21 @@ class ChunkedFile {
             return Result.failure(ChunkedFileError.fileHandle(error))
         }
     }
+
+    func fetchFileSize(
+        fileURL: URL
+    ) throws -> UInt64 {
+        guard let fileSize = try FileManager.default.attributesOfItem(atPath: fileURL.path)[FileAttributeKey.size] as? UInt64 else {
+            throw ChunkedFileError.invalidState("Cannot retrieve file size")
+        }
+
+        return fileSize
+    }
     
     /// Opens the internal file ahead of time. Calling this is optional, but it's available
     /// Calling this multiple times (on the same thread) will have no effect unless you also ``close`` it
     /// Throws if the file couldn't be opened
-    public func openFile(fileURL: URL) throws {
+    func openFile(fileURL: URL) throws {
         if fileHandle == nil {
             do {
                 guard let fileSize = try FileManager.default.attributesOfItem(atPath: fileURL.path)[FileAttributeKey.size] as? UInt64 else {
```

Comment:
I'd like to remove this side effect of opening the file entirely, but not sure if it would break anything else

Reply:
suggestion(non-blocking): While this side effect is still in place, consider refactoring to use your new `fetchFileSize()` function to avoid code duplication/maintenance concerns.

Reply:
I'd prefer to remove this side effect in this PR if possible, but if not good call will use the new method

Reply:
I don't think I understand what you have in mind by not opening the file.

What is the alternative to opening the file here? Would the caller then be keeping track of the `FileHandle` and read position and stuff? Or, like would you open and close the file for every read??

Reply:
Sorry this bit was unclear, the side effect in this thread refers to setting the `ChunkedFile.fileSize` property when opening the file. There's no intention to change the timing or logic of when the file is actually opened.